pkgs/development/compilers/gcc/12/default.nix

```diff
@@ -64,7 +64,10 @@ let majorVersion = "12";
         ../gnat-cflags-11.patch
         ../gcc-12-gfortran-driving.patch
         ../ppc-musl.patch
-      ] ++ optional (stdenv.isDarwin && stdenv.isAarch64) (fetchpatch {
+      ]
+      # We only apply this patch when building a native toolchain for aarch64-darwin, as it breaks building
+      # a foreign one: https://github.com/iains/gcc-12-branch/issues/18
+      ++ optional (stdenv.isDarwin && stdenv.isAarch64 && buildPlatform == targetPlatform) (fetchpatch {
```

Comment:
Minor nit: expressing a native toolchain would be `buildPlatform == hostPlatform && hostPlatform == targetPlatform`. I don't think you can easily violate it with `nixpkgs`. But I think it expresses intent in a more straightforward way.

Worth an extra test as it is a bit more strict than, say, `buildPlatform.config == hostPlatform.config && hostPlatform.config == targetPlatform.config`.

Reply:
Agreed. Thanks.

Reply:
Fixed, thanks.